### PR TITLE
Add to Cart Alert Behavior

### PR DESCRIPTION
**Pull request by andela-aatanda**
Resolves [1445](https://github.com/reactioncommerce/reaction/issues/1445) and [2042](https://github.com/reactioncommerce/reaction/issues/2042)

***Testing***

- Clone this branch and start up `reaction`
- Select a product and click `add to cart` button rapidly
- Note the spinner on rapid clicks
- Check that the cart drawer behaves the same way that it used to

### CODE REVIEW

**Comment by mikemurray** [comment]
If you are able to, can you convert this to React?

**Comment by mikemurray** [comment]
Here's the ticket that this may be related to: https://github.com/reactioncommerce/reaction/issues/2042

**Comment by andela-aatanda** [comment]
Okay, I'll do that.

**Comment by zenweasel** [comment]
A little nit-picky but there's a weird size change between the spinner and non spinner boxes.

add_to_cart

**Comment by zenweasel** [comment]
Some weirdness with the titles there
basic_reaction_product

**Comment by zenweasel** [comment]
Some weirdness with the delete from cart button. Every time I mouse over it there I am clicking on it. The screencap doesn't show it but the cursor turned into a hand when I mouseover it as well which is weird.

add_to_cart2

**Comment by zenweasel** [comment]
Seeing this error when I check out:

```
Exception from Tracker recompute function:
meteor.js?hash=27829e9…:930 Error: No such template: cartSubTotals
    at blaze.js?hash=f33d3df…:3212
    at Blaze.View.<anonymous> (spacebars.js?hash=ebf9381…:68)
    at blaze.js?hash=f33d3df…:1934
    at Function.Template._withTemplateInstanceFunc (blaze.js?hash=f33d3df…:3744)
    at blaze.js?hash=f33d3df…:1932
    at Object.Blaze._withCurrentView (blaze.js?hash=f33d3df…:2271)
    at viewAutorun (blaze.js?hash=f33d3df…:1931)
    at Tracker.Computation._compute (tracker.js?hash=9f8a0ce…:339)
    at new Tracker.Computation (tracker.js?hash=9f8a0ce…:229)
    at Object.Tracker.autorun (tracker.js?hash=9f8a0ce…:604)
```

**Comment by zenweasel** [comment]
Getting this error in the server console:

```
=> App running at: http://localhost:3000/
Exception in callback of async function: TypeError: Cannot read property 'shopId' of undefined
    at imports/plugins/included/shippo/server/methods/shippo.js:66:15
    at Array.map (native)
    at ratesParser (imports/plugins/included/shippo/server/methods/shippo.js:48:22)
    at [object Object].methods.shippo/getShippingRatesForCart (imports/plugins/included/shippo/server/methods/shippo.js:412:29)
    at packages/check.js:130:16
    at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
    at Object.exports.Match._failIfArgumentsAreNotAllChecked (packages/check.js:129:41)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1734:18)
    at packages/ddp-server/livedata_server.js:1650:18
    at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
    at [object Object]._.extend.apply (packages/ddp-server/livedata_server.js:1649:45)
    at [object Object]._.extend.call (packages/ddp-server/livedata_server.js:1592:17)
    at packages/kadira:flow-router-ssr/server/plugins/ssr_data.js:67:23
    at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
    at Object.Meteor.(anonymous function) [as call] (packages/kadira:flow-router-ssr/server/plugins/ssr_data.js:66:44)
    at getShippingRates (imports/plugins/included/shippo/server/hooks/rates.js:55:34)
    at server/api/hooks.js:53:14
    at Array.reduce (native)
    at Object.Hooks.Events.run (server/api/hooks.js:52:22)
    at [object Object].methods.shipping/getShippingRates (server/methods/core/shipping.js:81:18)
    at packages/check.js:130:16
    at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
```

**Comment by zenweasel** [comment]
If I add shipping I get weirdness in the cart drawer and more errors in the console

reaction_and_taxcalc_js_-_reaction_-____projects_js_reaction_

**Comment by zenweasel** [comment]
This is pretty close but make sure some of the subtle things like the spacing between the number and the product name are the same. (original on the left, new on the right)

add_to_cart_new_jpg_and_reaction

**Comment by andela-aatanda** [comment]
@zenweasel I've fixed the above reviews and the spinner size has been bumped up so as to have approximately same effect on the `cart-alert` size as the `cart-alert-text`

**Comment by zenweasel** [comment]
Products disappear in cart drawer as more narrow widths. Compare to current version where they shrink.

add_to_cart_width

**Comment by zenweasel** [comment]
This spacing, specifically pointed out in my previous comments, has not been fixed.

basic_reaction_product

File: imports/plugins/core/checkout/client/components/emptyCartDrawer.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { PropTypes } from "react";
+
+const EmptyCartDrawer = ({ keepShopping }) => {
+  return (
+    <div className="cart-drawer" id="cart-drawer">
+      <div className="cart-drawer-empty">
+        <div className="row cart-drawer-empty-msg">
+          <h1 className="text-align">
+            <i className="fa fa-frown-o fa-lg" /><strong><p data-i18n="cartDrawer.empty">We're sad. Your cart is empty.</p></strong>
```
Comment (zenweasel) [inline imports/plugins/core/checkout/client/components/emptyCartDrawer.js]: This doesn't appear to be fixed still

**Comment by zenweasel** [comment]
@mikemurray I think this is ready for you to take a look at and merge if approved

**Pull request merged**